File: count_pdf.py
import fitz
import re

doc = fitz.open('/Users/dmitt/Desktop/QuantGuide_Web/QUANT_GUIDE[1].pdf')
text = ""
for page in doc:
    text += page.get_text()

q_matches = re.findall(r'Question (\d+):', text)
s_matches = re.findall(r'Solution to Question (\d+):', text)

print(f"Total questions found by regex: {len(q_matches)}")
print(f"Total solutions found by regex: {len(s_matches)}")

unique_qs = set(q_matches)
unique_ss = set(s_matches)
print(f"Unique questions: {len(unique_qs)}")
print(f"Unique solutions: {len(unique_ss)}")

max_q = max(int(q) for q in unique_qs) if unique_qs else 0
print(f"Max question number: {max_q}")

File: old_scripts/count_pdf.js
const fs = require('fs');
const pdfParse = require('pdf-parse');

async function count() {
  const buf = fs.readFileSync('/Users/dmitt/Desktop/QuantGuide_Web/QUANT_GUIDE[1].pdf');
  const data = await pdfParse(buf);
  const text = data.text;
  
  const qMatches = [...text.matchAll(/Question (\d+):/g)];
  const sMatches = [...text.matchAll(/Solution to Question (\d+):/g)];
  
  console.log(`Total questions found by regex: ${qMatches.length}`);
  console.log(`Total solutions found by regex: ${sMatches.length}`);
  
  const uniqueQs = new Set(qMatches.map(m => m[1]));
  const uniqueSs = new Set(sMatches.map(m => m[1]));
  
  console.log(`Unique questions: ${uniqueQs.size}`);
  console.log(`Unique solutions: ${uniqueSs.size}`);
  
  let maxQ = 0;
  for (const q of uniqueQs) {
      if (parseInt(q) > maxQ) maxQ = parseInt(q);
  }
  console.log(`Max question number: ${maxQ}`);
}
count().catch(console.error);
